### ISSUE TITLE
Use Push Notification Service like Telegram

### ISSUE THREAD

**Issue by Elendil211** [vector-im/element-android#2519]
Problem:
It is necessary to disable battery optimization in order to instantly get notifications of incoming messages.
This is complicated and (as far as I know) uses more battery than necessary. Less tech-savy users will not know about this problem and just get frustrated.

Solution:
Telegram uses a message that is shown all the time, in order to be able to run a background service. The message tells more about the problem, if clicked upon:  https://github.com/Telegram-FOSS-Team/Telegram-FOSS/blob/master/Notifications.md
This works flawlessly and the notification can be hidden.

**Comment by notramo** [comment]
Element has this option too (permanent notification), but somehow it isn't permanent. And even with every battery optimisation disabled, it doesn't always run in the background.

**Comment by bahur142** [comment]
Element do not use permanent connection to the synapse server. 
XMPP like apps are needed this. Obviously telegram too. These are two different architectures.

**Comment by notramo** [comment]
Yes, these are different architectures. But I should be able to get a notification about every message. Maybe not instantly, but if I set sync period to 30 sec, then I should get the message no longer than 30 seconds after sending. **Sometimes I don't get the message until I open the app** despite I have set up everything correctly and running pure LineageOS.

**Comment by mitchmindtree** [comment]
Also running into this issue using Element via f-droid. I'm sitting with a friend testing Element notifications and can't seem to be able to find a configuration that allows for reliably receiving calls and messages in real-time, while on the other hand the Telegram-FOSS notification hack appears to work consistently.

Such a pity that Google puts not only FOSS apps in this position, but any Android device that cannot support Google Play Services (i.e. my Hisense E-ink phone, which can't even receive push notifications from any of the popular proprietary apps).

**Comment by Tuxicoman2** [comment]
Indeed, that the blocking point that keep me on XMPP because Conversations just works well on mobile phone.
This notification issue is there since years and I don't see how it can be solved...

**Comment by robsmith11** [comment]
It seems something like this is needed to get any notifications on my Huawei phone.  I don't get anything unless Element is open and in the foreground.
On my Samsung tablet, I get notifications fine when Element is closed and the screen is off.
